nixos/modules/services/web-apps/guacamole-server.nix

```diff
@@ -0,0 +1,131 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+let
+  cfg = config.services.guacamole-server;
+  guacamolePropertiesFormat = pkgs.formats.javaProperties { };
+  stateDir = "/run/guacd";
+  defaultGuacdUser = "guacd";
+  defaultGuacdGroup = "guacd";
+in
+{
+  options = {
+    services.guacamole-server = {
+      enable = lib.mkEnableOption (lib.mdDoc "Apache Guacamole Server");
+      package = lib.mkPackageOptionMD pkgs "guacamole-server" { };
+
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = defaultGuacdUser;
+        description = lib.mdDoc ''
+          User under which the Guacamole service runs.
+
+          ::: {.note}
+          If left as the default value this user will automatically be created
+          on system activation, otherwise it needs to be configured manually.
+          :::
+        '';
+      };
+
+      group = lib.mkOption {
+        type = lib.types.str;
+        default = defaultGuacdGroup;
+        description = lib.mdDoc ''
+          Group which runs Guacamole service.
+
+          ::: {.note}
+          If left as the default value this group will automatically be created
+          on system activation, otherwise you are responsible for
+          ensuring the group exists before the Guacamole service starts.
+          :::
+        '';
+      };
```

Comment:
We need a real user with a home directory because the rdp protocol (and probably other protocols) write in it. Should we investigate into using a dynamic user?